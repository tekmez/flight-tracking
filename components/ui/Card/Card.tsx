import { View } from "react-native";
import { Circle, ClipPath, Defs, Rect, Svg } from "react-native-svg";

type CardProps = {
  children: React.ReactNode;
};
export default function Card({ children }: CardProps) {
  return (
    <Svg height="151" width="100%">
      <Defs>
        <ClipPath id="clip">
          <Rect x="0" y="0" width="100%" height="151" rx="14" ry="14" />
          <Circle cx="0" cy="75.5" r="14" />
          <Circle cx="100%" cy="75.5" r="14" />
        </ClipPath>
      </Defs>
      <Rect
        x="0"
        y="0"
        width="100%"
        height="151"
        fill="white"
        clipPath="url(#clip)"
        rx={14}
        ry={14}
        stroke={"#E5E7EB"}
        strokeWidth={1}
      />
      <View className="p-4">{children}</View>
    </Svg>
  );
}
