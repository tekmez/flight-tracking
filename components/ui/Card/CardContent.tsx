import { View, Text } from "react-native";
import React from "react";
import UpArrowIcon from "@/assets/icons/UpArrowIcon";
import PlaneRoadIcon from "@/assets/icons/PlaneRoadIcon";
import UpDownIcon from "@/assets/icons/UpDownIcon";

type CardContentProps = {
  takeOffDate: string;
  departureCity: string;
  departureTime: string;
  landingDate: string;
  arrivalCity: string;
  arrivalTime: string;
};

const CardContent = ({
  takeOffDate,
  departureCity,
  departureTime,
  landingDate,
  arrivalCity,
  arrivalTime,
}: CardContentProps) => {
  return (
    <View className="flex-row items-center">
      <View className="ml-[3px]">
        <View className="flex-row items-center mb-2">
          <UpArrowIcon className="mr-1" />
          <Text className="font-sfPro text-[13px] leading-[22px] text-center text-gray-500">
            {takeOffDate}
          </Text>
        </View>
        <Text className="font-sfProBold text-[20px] leading-[25px] text-gray-900 mb-2">
          {departureCity}
        </Text>
        <Text className="font-sfProBold text-[13px] leading-[18px] text-gray-700">
          {departureTime}
        </Text>
      </View>
      <PlaneRoadIcon className="mx-auto" />
      <View className="mr-[3px] items-end">
        <View className="flex-row items-center mb-2">
          <UpDownIcon className="mr-1" />
          <Text className="font-sfPro text-[13px] leading-[22px] text-center text-gray-500">
            {landingDate}
          </Text>
        </View>
        <Text className="font-sfProBold text-[20px] leading-[25px] text-gray-900 mb-2">
          {arrivalCity}
        </Text>
        <Text className="font-sfProBold text-[13px] leading-[18px] text-gray-700">
          {arrivalTime}
        </Text>
      </View>
    </View>
  );
};

export default CardContent;
