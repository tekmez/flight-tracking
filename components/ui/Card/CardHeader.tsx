import { View, Text } from "react-native";
import React from "react";
import Badge from "../Badge";
import Airline from "../AirlineIcon";

type CardHeaderProps = {
  airline: string;
  duration: string;
  flightNumber: string;
};

const CardHeader = ({ airline, duration, flightNumber }: CardHeaderProps) => {
  const iconName = airline.split(" ")[0] as
    | "Turkish"
    | "Ryanair"
    | "Emirates"
    | "Lufthansa";
  return (
    <View className="flex-row justify-between items-center mb-4">
      <View className="flex-row items-center">
        <Airline name={iconName} />
        <Text className="font-sfProMedium text-[13px] leading-[18px] text-center text-gray-500 my-[3px] ml-2">
          {flightNumber} â€¢ {airline}
        </Text>
      </View>
      <Badge duration={duration} />
    </View>
  );
};

export default CardHeader;
