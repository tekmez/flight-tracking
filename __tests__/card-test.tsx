import React from "react";
import { render } from "@testing-library/react-native";
import Card from "@/components/ui/Card/Card";
import { Text } from "react-native";

describe("Card", () => {
  it("renders correctly", () => {
    const { toJSON } = render(
      <Card>
        <Text>Test Content</Text>
      </Card>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders children correctly", () => {
    const { getByText } = render(
      <Card>
        <Text>Child Component</Text>
      </Card>
    );
    expect(getByText("Child Component")).toBeTruthy();
  });
});
