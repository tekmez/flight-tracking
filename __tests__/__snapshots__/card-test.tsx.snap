// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<RNSVGSvgView
  bbHeight="151"
  bbWidth="100%"
  focusable={false}
  height="151"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 151,
        "width": "100%",
      },
    ]
  }
  width="100%"
>
  <RNSVGGroup
    fill={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGDefs>
      <RNSVGClipPath
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        name="clip"
      >
        <RNSVGRect
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          height="151"
          rx="14"
          ry="14"
          width="100%"
          x="0"
          y="0"
        />
        <RNSVGCircle
          cx="0"
          cy="75.5"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          r="14"
        />
        <RNSVGCircle
          cx="100%"
          cy="75.5"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          r="14"
        />
      </RNSVGClipPath>
    </RNSVGDefs>
    <RNSVGRect
      clipPath="clip"
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      height="151"
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      rx={14}
      ry={14}
      stroke={
        {
          "payload": 4293257195,
          "type": 0,
        }
      }
      strokeWidth={1}
      width="100%"
      x="0"
      y="0"
    />
    <RNSVGCircle
      cx="0"
      cy="75.5"
      fill={
        {
          "payload": 0,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      r="14"
      stroke={
        {
          "payload": 4293257195,
          "type": 0,
        }
      }
      strokeWidth="1"
    />
    <RNSVGCircle
      cx="100%"
      cy="75.5"
      fill={
        {
          "payload": 0,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
          "stroke",
          "strokeWidth",
        ]
      }
      r="14"
      stroke={
        {
          "payload": 4293257195,
          "type": 0,
        }
      }
      strokeWidth="1"
    />
    <View
      style={
        [
          {
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      <Text>
        Test Content
      </Text>
    </View>
  </RNSVGGroup>
</RNSVGSvgView>
`;
